pkg/target/regolib/src.rego

```diff
@@ -283,6 +302,11 @@ matches_namespaces(match) {
   not has_field(match, "namespaces")
 }
 
+# Always match cluster scoped resources
+matches_namespaces(match) {
```

Comment:
Keep in mind I'm still very new to rego, but I'm not sure I understand the 1:1 rule in this scenario.

I understand for functions which have return values it'll throw a runtime error if multiple functions evaluate to true. However doesn't seem to be the case for these functions since they don't return anything. It seems like it does a logical `or` between matching functions. So it shouldn't matter if multiple evaluate true.

Is this a best practice? Or am I missing something? (granted I still need to fix the logic)

Reply:
It's actually whether two rules have different return values (with undefined being ignored).

For example:

[true, undefined, undefined] -> true
[true, true, undefined] -> true
[true, true, false] -> runtime error

Rego actually uses trinary logic, like SQL, and inherits some of those quirks.

In this case, it looks like you;re right. Since true and undefined are the only possibilities for return values, it's impossible to have a runtime error. That being said, it's easier to validate that this is the case if the overloaded functions are disjoint, such that only one function at a time will ever return a value, and the rest are all undefined.

Reply:
Ah cool, thanks for explaining, that makes sense.

Reply:
Note that I found out [true, true, false] => true, ["yes", "no"] => runtime error.